### PR TITLE
Add multiple conda environments

### PR DESCRIPTION
**Pull request by dmentipl**
Addresses #1381

Add multiple conda environments in a new directory "envs" and test each in CI.

The existing conda environment file (environment.yml) is renamed environment_py3.6_iris2.2.yml and there is an additional Python 3.7 and Iris 2.4 environment. Both environments are tested via GitHub Actions.

Two tests for wind components required reduced precision to pass unit tests. The Python 3.7 environment required pinning pySTEPS to 1.3.2.

Testing:
 - [x] Ran tests and they passed OK

Note: this pull request contains the changes implemented in #1385. So this can be considered a draft until that is merged.

### CODE REVIEW

**Comment by dmentipl** [comment]
The issue with pySTEPS may be resolved in #1389.

File: envs/environment_py3.6_iris2.2.yml
Comment (tjtg) [inline envs/environment_py3.6_iris2.2.yml:24]: Bandit and safety are being removed from this environment, but they are used in tests.yml.
Suggest not removing these libraries as the bandit/pytest job in tests.yml is currently run in both environments.

**Review by tjtg — CHANGES_REQUESTED** [review]
This looks good overall. I've added comments about a few inconsistencies which are worth cleaning up before this gets merged.

File: .github/workflows/tests.yml
Comment (dmentipl) [reply .github/workflows/tests.yml]: After wrangling with GitHub Actions, I've addressed this. However, there still seems to be issues with the cache not being hit, so it runs the conda env update step, but then when it attempts to save the cache it says the cache already exists. I've tried resetting the cache with the `CACHE_NUMBER` variable. It doesn't seem to be working

File: .github/workflows/tests.yml
Comment (dmentipl) [reply .github/workflows/tests.yml]: I've addressed this by adding a step "conda info" that prints the conda info and packages.

File: envs/environment_py3.6_iris2.2.yml
Comment (dmentipl) [reply envs/environment_py3.6_iris2.2.yml]: I've fixed this.

File: .github/workflows/tests.yml
Comment (dmentipl) [reply .github/workflows/tests.yml]: However, this has thrown up an issue with pylint and Python 3.7. Possibly related to: https://github.com/PyCQA/pylint/issues/3318

We may have to pin pylint and astroid.

Comment (dmentipl) [reply .github/workflows/tests.yml]: Re: the caching issue, see the [Action output](https://github.com/dmentipl/improver/runs/1598935961?check_suite_focus=true#step:13:1).

Comment (dmentipl) [reply .github/workflows/tests.yml]: The caching issue seems to be resolved: see [here](https://github.com/dmentipl/improver/actions/runs/441595601).

With regards to the Pylint issue, I've disabled running in the Python 3.7 environment for now.

**Review by tjtg — DISMISSED** [review]
I'm happy with the updated branch here - looks like it's addressed all the items I mentioned previously and the tests are passing on github actions with both environments.
I've run the acceptance test suite with both 3.6/2.2 and 3.7/2.4 environments and the test failures match those already identified in issue #1397 and PR #1398.

File: .github/workflows/tests.yml
Comment (btrotta-bom) [inline .github/workflows/tests.yml:124]: In stage, we allow bandit and safety to return failure, but the repo has been configured so that these failures don't block merging. (See https://github.com/MetOffice/stage/pull/35#pullrequestreview-553432072). Maybe we should do same here to be consistent? CC @owena11

**Review by btrotta-bom — COMMENTED** [review]
This looks fine to me, I just had one question/comment about whether we should allow security checks to return failure.

File: .github/workflows/tests.yml
Comment (dmentipl) [reply .github/workflows/tests.yml:124]: Thanks for the review @btrotta-bom. This change was originally implemented in #1385. I'm happy to make that change but maybe it could be in a separate PR. This one is already getting large.

**Review by tjtg — APPROVED** [review]

**Pull request merged**
